src/lib.rs

```diff
@@ -300,15 +300,18 @@ impl Publisher for GooglePublisher {
                 // find the first item from the returned vector
                 response
                     .message_ids
-                    .ok_or(PublishError::InvalidResponseNoMessageId(not_found))
+                    .ok_or_else(|| PublishError::InvalidResponseNoMessageId(not_found))
                     .map(|v| v.into_iter().next())
                     .transpose()
-                    .unwrap_or(Err(PublishError::InvalidResponseNoMessageId(not_found)))
+                    .unwrap_or_else(|| Err(PublishError::InvalidResponseNoMessageId(not_found)))
```

Comment:
In these situations `unwrap_or` is perfectly fine. `unwrap_or_else` only matters when you have non-trivial error construction overhead, which in this case is not true.

Reply:
I agree. Going to remove and disable this clippy error.